sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -473,7 +473,7 @@ identifierComment
 
 relationPrimary
     : tableIdentifier sample? (AS? strictIdentifier)?      #tableName
-    | '(' queryNoWith ')' sample? (AS? strictIdentifier)?  #aliasedQuery
+    | '(' queryNoWith ')' sample? (AS? strictIdentifier)   #aliasedQuery
```

Comment:
I also have this question when I change this.

MySQL supports `select 1 from (test);` but doesn't support `select 1 from (test) as a ;`

Postgres doesn't support both syntax.

Hive supports `select * from (test);` but doesn't support `select * from (test) as a;`.

Seems aliased relation is not commonly supported. So I leave it untouched.